geospatial/src/Dem_TEST.cc

```diff
@@ -282,3 +282,34 @@ TEST_F(DemTest, LunarDemLoad)
   EXPECT_NEAR(dem.WorldWidth(), 80.0417, 1e-2);
   EXPECT_NEAR(dem.WorldHeight(), 80.0417, 1e-2);
 }
+
+TEST_F(DemTest, LargeVRTWithLimits)
+{
+  // Load a large VRT DEM using GDAL but set limits on the size.
+  common::Dem dem;
+  dem.SetXSizeLimit(100);
+  dem.SetYSizeLimit(50);
+  auto const path = common::testing::TestFile("data", "ap_srtm1.vrt");
+  auto const res = dem.Load(path);
+  EXPECT_EQ(res, 0);
+}
+
+TEST_F(DemTest, LargeVRTWithVSIZIPAndLimits)
+{
+  // Load a large vzizip VRT DEM using GDAL but set limits on the size.
+  common::Dem dem;
+  dem.SetXSizeLimit(100);
+  dem.SetYSizeLimit(50);
+  auto const path = common::testing::TestFile("data", "ap_srtm1.zip");
+  auto const res = dem.Load("/vsizip/" + path + "/ap_srtm1.vrt");
+  EXPECT_EQ(res, 0);
+}
+
+TEST_F(DemTest, LargeVRTWithoutLimitsThrows)
+{
+  // Load a large VRT DEM without limits.
+  common::Dem dem;
+  auto const path = common::testing::TestFile("data", "ap_srtm1.vrt");
+  // We expect not to be able to allocate the data, so ensure we throw instead of segfault.
+  EXPECT_THROW(dem.Load(path), std::bad_alloc);
```

Comment:
Is this actually causing `Dem::Load` to allocate more memory than available? I'm not sure if we want to do this in a unit test if it might badly affect other things running on CI.

Reply:
CI is failing on macOS with a timeout so the behavior may not be the same on all platforms, so I'd prefer we use `GTEST_SKIP` as you suggested.